test/Microsoft.AspNetCore.Blazor.E2ETest/Tests/EventTest.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using BasicTestApp;
+using Microsoft.AspNetCore.Blazor.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Blazor.E2ETest.Infrastructure.ServerFixtures;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Interactions;
+using OpenQA.Selenium.Support.UI;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Blazor.E2ETest.Tests
+{
+    public class EventTest : BasicTestAppTestBase
+    {
+        public EventTest(
+            BrowserFixture browserFixture,
+            DevHostServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+            Navigate(ServerPathBase, noReload: true);
+        }
+
+        [Fact] // Right now we're limited to bubbling events only, so no support for onfocus/onblur
+        public void FocusEvents_CanTrigger()
+        {
+            MountTestComponent<FocusEventComponent>();
+
+            var input = Browser.FindElement(By.Id("input"));
+
+            var output = Browser.FindElement(By.Id("output"));
+            Assert.Equal(string.Empty, output.Text);
+
+            // Focus the target, verify onfocusin is fired
+            input.Click();
+
+            Assert.Equal("onfocusin,", output.Text);
+
+            // Focus something else, verify onfocusout is also fired
+            var other = Browser.FindElement(By.Id("other"));
+            other.Click();
+
+            Assert.Equal("onfocusin,onfocusout,", output.Text);
+        }
+
+        [Fact]
+        public void MouseOverAndMouseOut_CanTrigger()
+        {
+            MountTestComponent<MouseEventComponent>();
+
+            var input = Browser.FindElement(By.Id("mouseover_input"));
+
+            var output = Browser.FindElement(By.Id("output"));
+            Assert.Equal(string.Empty, output.Text);
+
+            var other = Browser.FindElement(By.Id("other"));
+
+            // Mouse over the button and then back off
+            var actions = new Actions(Browser)
+                .MoveToElement(input)
+                .MoveToElement(other);
```

Comment:
I'm far from a webdriver expert but this seemed to be what they recommend for general interactivity scripting. 

We also have the ability to run javascript to trigger events, but I wanted to avoid that until its absolutely necessary because that seems like a 'less good' way to test.

Reply:
Oh, well, I should have read this earlier. Just realised I've also implemented similar E2E events coverage in #722. Sorry for duplicating your work! I even used the same filename 😞 

I guess it would be good to merge what we've done together. Your test has more coverage of the mouse events, whereas mine has more coverage of the bubbling behaviors.